libgnucash/core-utils/gnc-glib-utils.c

```diff
@@ -351,3 +351,33 @@ gnc_g_list_stringjoin (GList *list_of_strings, const gchar *sep)
 
     return retval;
 }
+
+gboolean
+gnc_list_length_cmp (GList *lst, size_t len,
+                     gboolean less, gboolean same, gboolean more)
```

Comment:
That's ugly API that shouldn't be public, so `static inline boolean list_length_cmp_internal`.